skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+        is unknown. The location may begin or end at some points outside
+        the specified coordinates. This accommodates the location format [1]_
+        of INSDC.
+
+        References
+        ----------
+        .. [1] ftp://ftp.ebi.ac.uk/pub/databases/embl/doc/FT_current.html#3.4.3
+        '''
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve boundaries from.'
+                               'dropped Interval object.')
+        return self._boundaries
+
+    @boundaries.setter
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self, value):
+        '''Set ``boundaries``.
+
+        The ``value`` should be ``None`` (meaning all boundaries are ``True``)
+        or an iterable matching ``self.locations``.'''
+        if self.dropped:
+            raise RuntimeError('Cannot change boundaries to dropped '
+                               'Interval object.')
+        nl = len(self.locations)
+        if value is None:
+            self._boundaries = [(True, True)] * nl
+        else:
+            try:
+                value = list(value)
+            except:
+                raise TypeError(
+                    'Cannot give a non-iterable (%r) to `boundaries`.' % value)
+
+            nb = len(value)
+            if nb != nl:
+                msg = ('The length of boundaries (%d) needs '
+                       'to be equal to the length of locations (%d).')
+                raise ValueError(msg % (nb, nl))
+
+            for boundary in value:
+                _assert_valid_boundary(boundary)
+            self._boundaries = value
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def locations(self):
+        '''The coordinates of the interval feature.
+
+        It should be a list of tuples of int pair. Each tuple stores
+        the start and end coordinates of a span of the interval feature.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve locations from dropped Interval object.')
+        return self._locations
+
+    @locations.setter
+    @experimental(as_of='0.5.0-dev')
+    def locations(self, value):
+        '''Set ``locations``.
+
+        WARNING: setting ``locations`` will reset ``boundaries`` value to True.
+        This is not totally surprising because it is justifiable your old
+        boundaries don't fit the new locations.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot change locations to dropped Interval object.')
+        # check iterability
+        try:
+            value = list(value)
+        except:
+            raise TypeError(
+                'Cannot give a non-iterable (%r) to `locations`.' % value)
+        # check it is not empty
+        if len(value) == 0:
+            raise ValueError(
+                'Cannot give empty `location`.')
+        # check each contiguous span is in right format
+        for location in value:
```

Comment:
Validation of `locations` doesn't account for the following cases:

- duplicate location tuples
- location tuples that are in `(end, start)` orientation instead of `(start, end)`

Do these cases need to be accounted for, or will the `IntervalMetadata` object (and `IntervalTree`) function correctly as-is with this type of data?